### PR TITLE
fix: stub out msal methods on non-supported platforms

### LINKED ISSUE #2484: [Authentication][MSAL] API surface differs between tfms
As of [this](https://github.com/unoplatform/uno.extensions/commit/14979905a3a4c35a2f72a546d3471ffa32da6bb9), we now need to wrap the MSAL builder calls with `UNO_EXT_MSAL`. Like we are doing in the [TestHarness](https://github.com/unoplatform/uno.extensions/blob/fbe8ccf3f18eef69b5acff868e1cc38d8151f5c9/testing/TestHarness/TestHarness.Shared/Ext/Authentication/Msal/MsalAuthenticationMultiHostInit.cs#L106).

We should document the platforms that are/are not supported and highlight the usage of `UNO_EXT_MSAL` in the [MSAL How-to docs](https://github.com/unoplatform/uno.extensions/blob/fbe8ccf3f18eef69b5acff868e1cc38d8151f5c9/doc/Learn/Authentication/HowTo-MsalAuthentication.md)

### PR DESCRIPTION
**Pull request by kazo0**
closes #2484

### CODE REVIEW

File: src/Uno.Extensions.Authentication.MSAL/Uno.Extensions.Authentication.MSAL.WinUI.csproj [hunk 1]
```diff
@@ -63,8 +61,8 @@
 
 	<ItemGroup>
 		<PackageReference Include="Microsoft.Extensions.Localization.Abstractions" />
-		<PackageReference Include="Microsoft.Identity.Client" Condition="'$(IsMSALSupported)'=='true'" />
-		<PackageReference Include="Microsoft.Identity.Client.Extensions.Msal" Condition="'$(IsMSALSupported)'=='true'" />
+		<PackageReference Include="Microsoft.Identity.Client" />
```
Comment (nickrandolph) [inline src/Uno.Extensions.Authentication.MSAL/Uno.Extensions.Authentication.MSAL.WinUI.csproj:64]: We should double check that these dependencies don't cause compilation errors (in Release) with any of the Uno targets

**Review by nickrandolph — APPROVED** [review]

File: src/Uno.Extensions.Authentication.MSAL/MsalAuthenticationBuilderExtensions.cs [hunk 2]
```diff
@@ -81,6 +86,9 @@ Action<StorageCreationPropertiesBuilder> store
 		string[] scopes
 		)
 	{
+#if !UNO_EXT_MSAL
+		return builder;
```
Comment (carldebilly) [inline src/Uno.Extensions.Authentication.MSAL/MsalAuthenticationBuilderExtensions.cs:90]: Should it raise a warning somehow?

**Review by carldebilly — APPROVED** [review]

File: src/Uno.Extensions.Authentication.MSAL/MsalAuthenticationBuilderExtensions.cs (hunk 2 above)
Comment (kazo0) [reply src/Uno.Extensions.Authentication.MSAL/MsalAuthenticationBuilderExtensions.cs:90]: I'm not sure how we should do that? I don't really have access to a logger here. Through an attribute maybe?

**Comment by kazo0** [comment]
@mergifyio backport release/stable/4.2

**Pull request merged**